### ISSUE TITLE
gen-variable should lower-case the content of xml:lang in strings file

### ISSUE THREAD

**Issue by simonbate** [dita-ot/dita-ot#2226]
A variables file in `customization.dir` with an upper case locale (such as `de_DE.xml`, as suggested in the `vars/README.txt` file) will never be matched in `dita-utilities.xsl` (findString). This is because the xml:lang attribute set in the strings.`*.xml` contains the literal filename portion, with no lower-casing, but the predicate in the `findString` template does an exact compare (`[@xml:lang = $l]`). 

To make this bullet-proof, `org.dita.dost.pdf2.VariableFileTask` should lower-case the filename string when it generates the `xml:lang` attribute. 

We've been able to work around this issue by making sure we use lowercase for the filename, but with the `README.txt` suggesting that the locale could be uppercase, it may trap others.

**Comment by robander** [comment]
Marking as a bug for the moment. I think there's a duplicate issue open somewhere (I ran into the same thing last November or December). I'm not sure if the correct "fix" at this point is to lower case values in the code, to update the README, or (probably) both.

**Comment by infotexture** [comment]
The ReadMe files in `org.dita.pdf2` were removed in 05e7b0d for #1860.

Relevant content is being added to docs for 2.3 with dita-ot/docs#69.

Most of the variable files no longer include the locale in the filename, so `de_DE.xml` is now `de.xml`. There are only 2 cases left where the locales are present at all, so it looks like this would only affect `pt_BR.xml` & `zh_CN.xml` at the moment.

**Comment by robander** [comment]
Yeah - I ran into it when working on support for those two last fall. Today I'm working on `zh-TW` so I expect that should also be one we ship in 2.3. Although it's worth noting - it's certainly possible that somebody could use different strings in their custom code for British English `en-GB`, or Swiss German `de-CH`, so it at least has the potential to hit extensions more than the base code.)

**Comment by robander** [comment]
I think this is part of a general regression, coming from a change in string files (probably the point when strings were updated to pull from both the common files, the PDF files, and any others). I've realized that strings for "zh-CN" no longer work for XHTML either, though they used to a few releases ago.

**Comment by robander** [comment]
Or ... now starting to wonder if something is wrong in my test. I may have been wrong about the XHTML failure.

**Comment by robander** [comment]
The existing strings files in PDF are declared in
`cfg/common/vars/strings.xml`

That file includes the following two lines (there is a PR open adding zh-TW):
```
<lang filename="pt_BR.xml" xml:lang="pt-BR"/>
<lang filename="zh_CN.xml" xml:lang="zh-CN"/>
```

I noticed that the zh_CN strings were not working - every string generated a "cannot find match" error. Similarly the pt-BR strings did not work, but I did not get errors, because it looks like they fell back to "pt".

I assume both of these are due to the reported bug. The quickest fix would be to change the case of these in `strings.xml` - they work with `pt-br`, `zh-cn`, and `zh-tw`, but it would be better if any value worked (particularly since nn-NN is a common convention).

**Comment by robander** [comment]
Commit 500761b at least partially addresses this by fixing the variable files we ship - using lower case in `org.dita.pdf2/cfg/common/vars/strings.xml` means shipped variables will now be found for all supported languages. I'd still give :+1: to a more general fix that lets variables be found regardless of case in the string integration file + in the source file.

**Comment by robander** [comment]
Finally have a more general fix for this in 2.5.4.
